src/Estimators/EstimatorManagerCrowd.h

```diff
@@ -86,8 +86,9 @@ class EstimatorManagerCrowd
   ///total weight accumulated in a block
   RealType block_weight_;
 
+  UPtr<ScalarEstimatorBase> main_estimator_;
   ///estimators of simple scalars
-  std::vector<std::unique_ptr<EstimatorType>> scalar_estimators_;
+  std::vector<std::unique_ptr<ScalarEstimatorBase>> scalar_estimators_;
 
   std::vector<std::unique_ptr<OperatorEstBase>> operator_ests_;
```

Comment:
We will need to do some refactoring in a **separate** PR. main_estimator_, scalar_estimators_, operator_ests_ should be const unique_ptr as they are being settled in the constructor and never intended to be changed throughout its life time.

EstimatorManagerCrowd should not be a friend of EstimatorManagerNew and EstimatorManagerNew should have APIs to return clones of these collection of estimators.